dali/pipeline/pipeline.cc

```diff
@@ -516,6 +516,11 @@ void Pipeline::PrepareOpSpec(OpSpec *spec) {
     .AddArg("num_threads", num_threads_)
     .AddArg("device_id", device_id_)
     .AddArgIfNotExisting("seed", seed_[current_seed_]);
+  string dev = spec->GetArgument<string>("device");
+  if (dev == "cpu" || dev == "mixed")
+    spec->AddArg("cpu_prefetch_queue_depth", prefetch_queue_depth_.cpu_size);
+  if (dev == "gpu" || dev == "mixed")
```

Comment:
Sth is strange is these two conditionals. I mean, one of `dev == "mixed"` is redundant.

Reply:
What would you write?

Reply:
Either
```
  if (dev == "cpu" || dev == "mixed")
    [...]
  if (dev == "gpu")
    [...]
```
or
```
  if (dev == "cpu")
    [...]
  if (dev == "gpu" || dev == "mixed")
    [...]
```
depending on how would you like the "mixed" to be used
Also,
```
else {
  DALI_FAIL("Incorrect device name");
}
```
might be nice for an (in)sanity check ;)

Reply:
But logic-wise, the conditions you wrote are not the same as what I wrote

Reply:
I want `dev == "mixed"` to satisfy both conditions

Reply:
I am not sure to get what you are suggesting here 😛

Reply:
It's "get the cpu queue depth if the op is cpu or mixed AND get the gpu queue depth if the op is gpu or mixed". As Serge said, we want the mixed op to satisfy both conditions

About the sanity check, I believe it should be done at the very beginning of the pipeline creation, not here.